codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/customization/Route53Customizations.java

```diff
@@ -110,7 +110,7 @@ private void writeHostedZoneIDInputSanitizer(
                             writer.addUseImports(SmithyGoDependency.STRINGS);
                             for (MemberShape member : hostedZoneIDMembers) {
                                writer.openBlock("if i.$L != nil {", "}", member.getMemberName(), () -> {
-                                writer.write("values := strings.SplitN(*i.$L, \"/\", 2)", member.getMemberName());
+                                writer.write("values := strings.Split(*i.$L, \"/\")", member.getMemberName());
```

Comment:
We can simplify this and use [strings.LastIndex](https://pkg.go.dev/strings#LastIndex), and either use the value return or if `-1` fallback to `0`. The the write becomes`v := *i.$L[idx:]`. Removes the need for the slice of strings created.

```go
idx := strings.LastIndex(*i.$L, `/`) // using ` for string literals removes need for escaping in codegen too
if idx == -1 { idx = 0 }
v := *i.$L[idx:]
i.$L = &v
```